### PR TITLE
[Merged by Bors] - Add get_change_ticks method to EntityRef and EntityMut

### PR DESCRIPTION
**Pull request by TheRawMeatball**
Direct access to the change ticks is useful for integrating the reliable change detection with external stuff.

### CODE REVIEW

**Comment by mockersf** [comment]
As this is a little advanced, some doc comments would be nice 😃

File: crates/bevy_ecs/src/world/entity_ref.rs [hunk 1]
```diff
@@ -69,6 +69,20 @@ impl<'w> EntityRef<'w> {
         }
     }
 
+    #[inline]
+    pub fn get_change_ticks<T: Component>(&self) -> Option<&'w ComponentTicks> {
+        // SAFE: entity location is valid and returned component is of type T
+        unsafe {
+            get_component_and_ticks_with_type(
```
Comment (cart) [inline crates/bevy_ecs/src/world/entity_ref.rs:76]: I know you're trying to reuse code here, but we don't need to retrieve the component for this operation. This seems like the sort of small inefficiency that we'd merge and then forget about. A theoretical `get_ticks` and the existing `get_component` and `get_component_with_ticks` methods probably have some refactoring potential to share code. They all need to "get the entity's offset in each storage". This seems like a relatively trivial refactor for Table storage, but Sparse Set storage would probably need some new get_with_dense_index_unchecked() apis.

**Comment by TheRawMeatball** [comment]
Closing as I no longer need this, might reopen if I do.

**Comment by TheRawMeatball** [comment]
Reopening because I need this

File: crates/bevy_ecs/src/world/entity_ref.rs [hunk 2]
```diff
@@ -524,6 +547,31 @@ unsafe fn get_component_and_ticks(
     }
 }
 
+#[inline]
+unsafe fn get_ticks(
+    world: &World,
+    component_id: ComponentId,
+    entity: Entity,
+    location: EntityLocation,
+) -> Option<*mut ComponentTicks> {
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/world/entity_ref.rs:556]: This should use the new pointer types that you added in #3001 now.

File: crates/bevy_ecs/src/storage/sparse_set.rs [hunk 3]
```diff
@@ -183,10 +183,10 @@ impl ComponentSparseSet {
     }
 
     #[inline]
-    pub fn get_ticks(&self, entity: Entity) -> Option<&ComponentTicks> {
+    pub fn get_ticks_ptr(&self, entity: Entity) -> Option<*mut ComponentTicks> {
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/storage/sparse_set.rs:186]: This should use the new pointer types that you added in #3001 now.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
I'm having a hard time seeing what users would do with this, but I'm open to being convinced.

In any case, now that #3001 is merged, we can use lifetimed pointers!

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**